### ISSUE TITLE
Request never finish if error is thrown in async function

### ISSUE THREAD

**Issue by cgarciae** [luizmineo/redstone.dart#83]
Here is a minimal case
```dart
@app.Route ('/testErro1')
@Encode()
testError1 () async
{
    var resp = await testError2();
    
    return resp;
}

@app.Route ('/testErro2')
@Encode()
testError2 () async
{
    throw new app.ErrorResponse(400, {"error" : "invalid id"});
    
    return new BoolResp()
        ..value = true;
}
```

**Comment by cgarciae** [comment]
I created this plugin to handle this
```
class Catch {
  
  const Catch();
  
}

void ErrorCatchPlugin(app.Manager manager) {
    
    
    manager.addRouteWrapper(Catch, (metadata, Map<String,String> pathSegments, 
                                    injector, app.Request request, 
                                    app.RouteHandler route) async {
    
        try
        {
            var result = route(pathSegments, injector, request);
            
            if (result is Future)
                return await result;
            else
                return result;
        }
        catch (e, s)
        {
            return new Resp()
                ..error = "$e $s";
        }
    
  }, includeGroups: true);
```
I love plugins! However, Redstone should handle this by default.

**Comment by luizmineo** [comment]
Thanks for reporting.

**Comment by luizmineo** [comment]
I've released v0.5.21 with a fix for this. Please confirm.

**Comment by cgarciae** [comment]
Worked :)
